### PR TITLE
CI: Add QEMU functional tests

### PR DESCRIPTION
**Pull request by spowelljr**

### CODE REVIEW

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ var runCorpProxy = detect.GithubActionRunner() && runtime.GOOS == "linux" && !ar
 // TestFunctional are functionality tests which can safely share a profile in parallel
 func TestFunctional(t *testing.T) {
 
-	profile := UniqueProfileName("functional")
+	profile := UniqueProfileName("func")
```
Comment (medyagh) [inline test/integration/functional_test.go:76]: why this change?

Comment (spowelljr) [reply test/integration/functional_test.go:76]: The directory path length is too long for QEMU

**Comment by spowelljr** [comment]
Ran this previously and it didn't run into the `dhcpd_leases` issue

https://github.com/spowelljr/minikube/actions/runs/6475594083/job/17582985059

`functional_qemu_macos : completed with 1 / 67 failures in 15 min 43 seconds`
